### PR TITLE
Permit creation of process to configure options

### PR DESCRIPTION
**Pull request by ramirovarandas**
Hi,

  Currently the `execute` method allows to run a command inside the container and returning a Symfony Process, however, it's not possible to adjust the process settings because it has been started inside the `execute` method.
  I might be mistaken, but even with the `Macroable` trait, we're still unable to configure options before the process start.
  This PR adds a method called `makeProcess` which will simply make everything that the `execute` method does, except running the process, so in the case the user will be able to make adjustments and manually run the process.
  Added a test to check the modifications to the process instance and also added documentation to README.
  Thanks a lot for this package.

### CODE REVIEW

**Comment by freekmurze** [comment]
Thank you for you work on this. I don't like the fact that `$fullCommand` is defined twice now. Maybe it's better to pass a prepared command and let `execute` use if

```
$instance->useCommand($command)->execute($string)
```

**Comment by ramirovarandas** [comment]
I see your point, but when you say  `$instance->useCommand($command)` it will return the `Process` instance, right?

**Comment by freekmurze** [comment]
>  Or do you want just to build the command and store in a property and let the user use execute() or makeProcess() ?

It seems to me that this would be a nice way to go about it.

**Comment by ramirovarandas** [comment]
Hi,

  I was making the changes to this idea, however it would break backwards compatiblity, as `execute()` would not receive a command anymore if it was passed on to `useCommand($command)`. Am I mistaken?

  In my changes the code is like:

```php
    private string $command;

    // ...

    /**
     * @param string|array $command
     */
    public function useCommand($command): self
    {
        if (is_array($command)) {
            $command = implode(';', $command);
        }

        $this->command = "echo \"{$command}\" | docker exec --interactive {$this->getShortDockerIdentifier()} bash -";

        return $this;
    }
    
    /**
     * @return \Symfony\Component\Process\Process
     */
    public function execute(): Process
    {
        $process = Process::fromShellCommandline($this->command);

        $process->run();

        return $process;
    }

    /**
     * @return \Symfony\Component\Process\Process
     */
    public function makeProcess(): Process
    {
        $process = Process::fromShellCommandline($this->command);

        return $process;
    }
```

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,4 @@ docs
 vendor
 coverage
 .phpunit.result.cache
-.idea
+.idea
```
Comment (freekmurze) [inline .gitignore:7]: Please remove this, ideally you add this to your global .gitignore

File: src/DockerContainer.php [hunk 2]
```diff
@@ -132,6 +132,10 @@ public function start(): DockerContainerInstance
 
         $process = Process::fromShellCommandline($command);
 
+        // increasing the timeout limit
```
Comment (freekmurze) [inline src/DockerContainer.php:135]: Why were these lines added?

File: src/DockerContainerInstance.php [hunk 3]
```diff
@@ -86,6 +86,24 @@ public function execute($command): Process
         return $process;
     }
 
+    /**
+     * @param string|array $command
+     *
+     * @return \Symfony\Component\Process\Process
+     */
+    public function makeProcess($command): Process
```
Comment (freekmurze) [inline src/DockerContainerInstance.php:94]: I like the added functionality, but not the implementation. The code of `execute` appears twice now in the class. I think it would be better to add a method `configureProcess` that accepts a callable.

That callable should be kept on `DockerContainerInstance` as an instance variable. In `execute` the created process can be passed to the kept callable.

Could you make this modification?

File: src/DockerContainer.php (hunk 2 above)
Comment (ramirovarandas) [reply src/DockerContainer.php:135]: Sorry, I'll have to open another PR for you, because I'm already using these modifications for my production code.
Let's cancel this PR and I'll open another using another branch. Really sorry about this.
